apps/test/unit/lib/kits/maker/MicroBitBoardTest.js

```diff
@@ -0,0 +1,71 @@
+import {expect} from '../../../../util/deprecatedChai';
+import MicroBitBoard from '@cdo/apps/lib/kits/maker/MicroBitBoard';
+import {MicrobitStubBoard} from './makeStubBoard';
+import sinon from 'sinon';
+
+describe('MicroBitBoard', () => {
+  let board;
+
+  beforeEach(() => {
+    // Construct a board to test on
+    window.SerialPort = {};
+    board = new MicroBitBoard();
+    board.boardClient_ = new MicrobitStubBoard();
+  });
+
+  afterEach(() => {
+    board = undefined;
+  });
+
+  describe(`connect()`, () => {
+    it('initializes a set of components', () => {
+      return board.connect().then(() => {
+        expect(Object.keys(board.prewiredComponents_)).to.have.length(6);
+        expect(board.prewiredComponents_.board).to.be.a('object');
+        expect(board.prewiredComponents_.ledMatrix).to.be.a('object');
+        expect(board.prewiredComponents_.tempSensor).to.be.a('object');
+        expect(board.prewiredComponents_.accelerometer).to.be.a('object');
+        expect(board.prewiredComponents_.buttonA).to.be.a('object');
+        expect(board.prewiredComponents_.buttonB).to.be.a('object');
+      });
+    });
+  });
+
+  describe(`enableComponents())`, () => {
+    it('triggers a component start call if there are prewired components', () => {
+      return board.connect().then(() => {
+        let accelerometerSpy = sinon.spy(
```

Comment:
Is the accelerometer the only component that gets started by enableComponents() or just an example of one?

Reply:
This was a shorthand way to check if a different function that I couldn't watch was called. enableComponents() calls enableMicroBitComponents() which enables both the accelerometer and the thermometer. EnableMicroBitComponents() is tested to ensure both components are enabled when called. enableComponents() only calls triggers a call to enableMicroBitComponents. I used a check for whether enableMicroBitComponents() had been triggered (waiting for a call on one of the components it will trigger) to determine whether enableComponents() had successfully triggered enableMicroBitComponents().

Reply:
I think that makes sense to me, but could you add a comment explaining that inline?